integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py

```diff
@@ -0,0 +1,161 @@
+        """
+        Creates a new MongoDBAtlasDocumentStore instance.
+
+        This Document Store uses MongoDB Atlas as a backend (https://www.mongodb.com/docs/atlas/getting-started/).
+
+        :param mongo_connection_string: MongoDB Atlas connection string in the format:
+            "mongodb+srv://{mongo_atlas_username}:{mongo_atlas_password}@{mongo_atlas_host}/?{mongo_atlas_params_string}".
+            This can be obtained on the MongoDB Atlas Dashboard by clicking on the `CONNECT` button.
+            This value will be read automatically from the env var "MONGO_CONNECTION_STRING".
+        :param database_name: Name of the database to use.
+        :param collection_name: Name of the collection to use.
+        :param recreate_collection: Whether to recreate the collection when initializing the document store.
+        """
+        if collection_name and not bool(re.match(r"^[a-zA-Z0-9\-_]+$", collection_name)):
+            msg = f'Invalid collection name: "{collection_name}". It can only contain letters, numbers, -, or _.'
+            raise ValueError(msg)
+
+        resolved_connection_string = mongo_connection_string.resolve_value()
+        self.mongo_connection_string = mongo_connection_string
+
+        self.database_name = database_name
+        self.collection_name = collection_name
+        self.recreate_collection = recreate_collection
+
+        self.connection: MongoClient = MongoClient(
+            resolved_connection_string, driver=DriverInfo(name="MongoDBAtlasHaystackIntegration")
+        )
+        database = self.connection[self.database_name]
+
+        if self.recreate_collection and self.collection_name in database.list_collection_names():
+            database[self.collection_name].drop()
+
+        # Implicitly create the collection if it doesn't exist
+        if collection_name not in database.list_collection_names():
+            database.create_collection(self.collection_name)
+            database[self.collection_name].create_index("id", unique=True)
+
+        self.collection = database[self.collection_name]
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Utility function that serializes this Document Store's configuration into a dictionary.
+        """
+        return default_to_dict(
+            self,
+            mongo_connection_string=self.mongo_connection_string.to_dict(),
+            database_name=self.database_name,
+            collection_name=self.collection_name,
+            recreate_collection=self.recreate_collection,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "MongoDBAtlasDocumentStore":
+        """
+        Utility function that deserializes this Document Store's configuration from a dictionary.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["mongo_connection_string"])
+        return default_from_dict(cls, data)
+
+    def count_documents(self, filters: Optional[Dict[str, Any]] = None) -> int:
```

Comment:
This doesn't respect the protocol: https://github.com/deepset-ai/haystack/blob/main/haystack/document_stores/types/protocol.py#L38

Reply:
https://github.com/deepset-ai/haystack-core-integrations/pull/430